testcases/cloud_admin/modify_instance_type.py

```diff
@@ -0,0 +1,176 @@
+        self.name = name
+        self.cpus = cpus
+        self.mem = mem
+        self.disk = disk
+
+    def p(self):
+        print ("Name: " + self.name + " cpu: " + str(self.cpus) + " mem: "
+               + str(self.mem) + " disk: " + str(self.disk))
+
+class InstanceTypeOps(EutesterTestCase):
+    _known_types = []
+
+    def __init__(self, name="InstanceTypeOps", credpath=None, region=None, config_file=None, password=None, emi=None, zone=None,
+                  user_data=None, instance_user=None, **kwargs):
+
+        super(InstanceTypeOps, self).__init__(name=name)
+        self.get_args()
+        #self.show_args()
+        for kw in kwargs:
+            print 'Setting kwarg:'+str(kw)+" to "+str(kwargs[kw])
+            self.set_arg(kw ,kwargs[kw])
+        #self.show_args()
+        self.setuptestcase()
+        self.setup_parser()
+        #self.get_args()
+        # Setup basic eutester object
+        self.tester = Eucaops(config_file=self.args.config_file,
+                                  password=self.args.password)
+                                  # credpath=self.args.credpath)  # <- credpath disabled since not functioning for these commands
+
+        self.PopulateInstanceTypes()
+        #print "Now printing _my_types after populating:"    # DEBUG
+        #for i in InstanceTypeOps._known_types:
+            #i.p()
+
+    def clean_method(self):
+        pass
+
+    def check_output_type(self, arg=None):
+        """
+        :param arg: argument must be a list or it will assert
+        :return: assert if not list
+        """
+        assert type(arg) != None, "Output from self.sys() empty!"
+        try:
+            not isinstance(arg, basestring)
+        except TypeError:
+            AssertionError("Output from self.sys() should be list!")
+        #print "Output type checks out correctly!"    # DEBUG
+
+    def DescribeInstanceTypes(self, item=None):
+
+        if item is None:
+            cmd = "source " + self.tester.credpath + "/eucarc && /usr/bin/euca-describe-instance-types"
+        else:
+            cmd = "source " + self.tester.credpath + "/eucarc && /usr/bin/euca-describe-instance-types " + item
+
+        print "in DescribeInstanceTypes function Running " + cmd
+
+        output = self.tester.sys(cmd)
+
+        self.check_output_type(output)
+
+        return output
+
+    def ModifyInstanceType(self, item=None):
+
+        if item is None:
+            raise ValueError("Incorrect number of arguments, must pass cmd to ModifyInstanceType()")
+        else:
+            cmd = "source " + self.tester.credpath + "/eucarc && /usr/bin/euca-modify-instance-type " + ' '.join(item)
+
+
+        print "in ModifyInstanceType function Running " + cmd
+        output = self.tester.sys(cmd)
+        self.check_output_type(output)
+        return output
+
+    def PopulateInstanceTypes(self):
+        print "Populating cluster Instance Types..."
+        output = self.DescribeInstanceTypes()
```

Comment:
These calls should be implementable via API calls rather than parsing output of a CLI command. I know @bigschwan has done this we other calls, you should sync with him on how that is implemented.